sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -129,6 +129,9 @@ public RequestFailedException(Response response, Exception? innerException, Requ
         }
 
         /// <inheritdoc />
+#if NET6_0_OR_GREATER
+        [Obsolete(DiagnosticId = "SYSLIB0051")]
```

Comment:
See: [SYSLIB0051: Legacy serialization support APIs are obsolete](https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051)

Reply:
Are there other ways we could handle this one?  Do we want obsoleted code in Azure.Core?

Reply:
We can remove the APIs, but that would be breaking since they're public. Alternatively, we can suppress this warning for core libraries in the csproj files: https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0051#suppress-a-warning